examples/glTFViewer/glTFViewer.cpp

```diff
@@ -271,15 +271,18 @@ namespace {
         }
         auto inputState = builder.GetResult();
 
-        auto bindGroupLayoutBuilder = device.CreateBindGroupLayoutBuilder()
-            .SetBindingsType(nxt::ShaderStageBit::Vertex, nxt::BindingType::UniformBuffer, 0, 1)
-            .Clone();
+        nxt::BindGroupLayout bindGroupLayout;
         if (hasTexture) {
-            bindGroupLayoutBuilder
-                .SetBindingsType(nxt::ShaderStageBit::Fragment, nxt::BindingType::Sampler, 1, 1)
-                .SetBindingsType(nxt::ShaderStageBit::Fragment, nxt::BindingType::SampledTexture, 2, 1);
+            bindGroupLayout = utils::MakeBindGroupLayout(device, {
+                { 0, nxt::ShaderStageBit::Vertex, nxt::BindingType::UniformBuffer, 1 },
+                { 1, nxt::ShaderStageBit::Fragment, nxt::BindingType::Sampler, 1 },
+                { 2, nxt::ShaderStageBit::Fragment, nxt::BindingType::SampledTexture, 1 },
+            });
+        } else {
+            bindGroupLayout = utils::MakeBindGroupLayout(device, {
+                { 0, nxt::ShaderStageBit::Vertex, nxt::BindingType::UniformBuffer, 1 },
```

Comment:
The code duplication here is a little unfortunate. Can't think of a clean way to fix it, though. (I'm no std::initializer_list expert, but I'm assuming there's no way to dynamically concatenate the optional ones.) Maybe a per-binding boolean which says whether to use that binding or not? Or use an array size of zero to signal an unused binding, and zero out the unused bindings' array sizes? Neither seems great, though.